### PR TITLE
Fb steph bug fixes

### PR DESCRIPTION
**Pull request by Cell00phane**
This PR introduces several bug fixes and/or improvements:
- Starting time of events is formatted depending on whether it's occuring today, tomorrow or later
- integrate the user settings into the existing code base
- Avoid null number of participants when displaying my events
- Sync local cache's events with that of the remote in EventListFragment (to avoid displaying obsolete events)
- Add progress dialog on EventActivity until all the info from db is fetched (Should integrate to whole app)

### CODE REVIEW

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/HelperFunctions.kt [hunk 1]
```diff
@@ -244,6 +245,40 @@ object HelperFunctions {
...
+    /**
+     * Takes date instance with time and another date and returns the format as follows:
+     * - if dateTime occurs the same day we return (e.g. "Today at 07:36")
+     * - if dateTime occurs the day after the other date we return (e.g. "Tomorrow at 8:30"
+     * - else return the date and time (e.g. July 26 at 23:00)
+     * @param dateTime the LocalDateTime instance we're trying to format
+     * @param other the other date, which is just a date without time, so we can compare days
+     * @return the formatted date time with respect to the other date
+     */
+    fun formatDateTimeWithRespectToAnotherDate(dateTime: LocalDateTime?, other: LocalDate): String {
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/helper/HelperFunctions.kt:257]: This is nice :)

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/MainActivity.kt [hunk 2]
```diff
@@ -74,18 +85,34 @@ class MainActivity : AppCompatActivity() {
...
+                    if (userSettingsObservable.isNotEmpty() && userSettingsObservable[0].isSendingLocationOn) {
+                        HelperFunctions.getLoc(this).observeOnce { LatLng ->
+                            if (LatLng.value != null)
+                                currentDatabase.heatmapDatabase!!.setLocation(
+                                    LatLng.value,
+                                    userSettingsObservable
+                                )
+                        }
                     }
+                }
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/activity/MainActivity.kt:115]: Could you add one line of comment to explain what it does

**Review by Mathematou — APPROVED** [review]
Very nice ;)

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/MainActivity.kt (hunk 2 above)
Comment (Cell00phane) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/MainActivity.kt:115]: Yes will do. But basically this is the same code as before, just with the user settings fetched from local cache.

File: app/src/main/java/com/github/sdpteam15/polyevents/helper/HelperFunctions.kt [hunk 3]
```diff
@@ -244,6 +245,40 @@ object HelperFunctions {
...
+                formatted = "Tomorrow at $formatted"
+            } else {
+                val dateFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("MMMM dd")
+                // if not on the same day or day after, append the day and month as well
+                formatted = "${dateTime.format(dateFormatter)} at $formatted"
+            }
+        }
+        return formatted
+    }
+
```
Comment (AlessioVerardo) [inline app/src/main/java/com/github/sdpteam15/polyevents/helper/HelperFunctions.kt:281]: You can use the apply function from @Rygone

File: app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/EventListFragment.kt [hunk 4]
```diff
@@ -120,25 +117,66 @@ class EventListFragment : Fragment() {
...
+                }) {
+                eventLocalViewModel.delete(eventLocal)
+                eventsLocalCopy.remove(eventLocal)
+                eventsChanged = true
+            }
+        }
+
+        if (eventsChanged) {
+            eventsLocal.updateAll(eventsLocalCopy)
+            showToast("Events have been updated", context)
```
Comment (AlessioVerardo) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/EventListFragment.kt:160]: You can extract this into the string files

**Review by AlessioVerardo — APPROVED** [review]
Great code, as always :)

**Review by Feez2403 — APPROVED** [review]
great job, nice clear comments :)

**Pull request merged**